### PR TITLE
Updated/Corrected Single Use Algo/Examples

### PR DESCRIPTION
**Pull request by quartzjer**

### CODE REVIEW

**Review by selfissued — APPROVED** [review]
I'm going to merge this because I'm reviewing this spec with our engineering team in the morning.  Please nonetheless reply to my comments.  If you're busy, Jeremy, I'll create a PR addressing them after I hear your responses.

Thanks for doing this!

File: draft-jmiller-json-proof-algorithms.md [hunk 1]
```diff
@@ -130,45 +130,41 @@ The Single Use (SU) algorithm is based on composing multiple traditional JWS val
...
+### Using JWS
 
-The Parts List will always be an ordered array with a length of the total number of payloads plus one (for the protected header).  Each item in the list is an individual unencoded (binary) octet string of variable length.
+JSON Web Signatures are used to create the signature values used by the SU algorithm.  This allows an implementation to use an existing JWS library directly for all necessary cryptographic operations without requiring any additional primitives.
 
-### Using JWS
+Each individual JWS uses a fixed  protected header containing only the minimum required `alg` value.  Since this JWS protected header itself is the same for every JWS, it SHOULD be a static value in the form of `{"alg":"***"}` where `***` is the JWS asymmetric key algorithm being used.  This value is re-created by a Verifier using the correct algorithm value.
 
-The ephemeral key-pair is then used to generate an individual JWS for each item in the Parts List.
+If an implementation uses an alternative JWS protected header than the fixed value then a base64url encoded serialized form of that fixed header MUST be included as the `proof_header` value in the JWP protected header.
```
[the remark this answers is not in the document]
Comment (quartzjer) [reply draft-jmiller-json-proof-algorithms.md:143]: I am not aware of any need, but I decided to include it here as a placeholder should one arise.

File: draft-jmiller-json-proof-algorithms.md [hunk 2]
```diff
@@ -130,45 +130,41 @@ The Single Use (SU) algorithm is based on composing multiple traditional JWS val
...
 
-Each JWS is created with it's own protected header containing only the minimum required `alg` value.  Since the JWS protected header itself is not included in the resulting JWP, it must be a static value in the form of `{"alg":"***"}` where `***` is the JWS asymmetric key algorithm being used.  This value is re-created by a Verifier based on the `proof_jwk` included in the JWP protected header.
+### Protected Header
 
-The octet string of each Parts List item is used as the JWS body, creating a unique JWS for each item.
+The JWK of the ephemeral key MUST be included in the JWP protected header with the property name of `proof_jwk` and contain only the public values.
```
[the remark this answers is not in the document]
Comment (quartzjer) [reply draft-jmiller-json-proof-algorithms.md:147]: Yes, that's much better language for it :+1:

File: draft-jmiller-json-proof-algorithms.md [hunk 3]
```diff
@@ -130,45 +130,41 @@ The Single Use (SU) algorithm is based on composing multiple traditional JWS val
 
 ### Signer Setup
 
-To create a Single Use JWP the Signer must first generate a unique ephemeral key-pair using a JWS algorithm.  This key-pair will be used to sign the parts of a single JWP and then discarded.  The selected algorithm must be the same for both the ephemeral key and the Signer's public key.
+To create a Single Use JWP the Signer must first generate a unique ephemeral key-pair using a JWS algorithm.  This key-pair will be used to sign the parts of a single JWP and then discarded.  The selected algorithm must be the same for both the ephemeral key and the Signer's stable key.
```
[the remark this answers is not in the document]
Comment (quartzjer) [reply draft-jmiller-json-proof-algorithms.md:133]: Agreed!

File: draft-jmiller-json-proof-algorithms.md [hunk 4]
```diff
@@ -130,45 +130,41 @@ The Single Use (SU) algorithm is based on composing multiple traditional JWS val
...
 
 ### Selective Disclosure
 
-The Prover does not modify the JWP Proof value when presenting it to a Verifier, it will always contain the full set of JWS signature parts for each payload and the protected header.  Since this value is fixed for the JWP and always revealed, they should only be used and presented a single time to each Verifier in order for the Prover to remain unlinkable across multiple interactions.
+The Prover is able to modify the Proof value when presenting it to a Verifier, it will always contain the initial stable header signature part and is then followed by the ephemeral signature parts for each payload that is disclosed.  Non-disclosed payloads will not have their ephemeral signature value included (for example if the second and fifth payloads are hidden then the Prover's derived proof value would be `64 * (1 header signature + the 1st, 2nd, and 4th payload signatures) = 256 octets`).
```
[the remark this answers is not in the document]
Comment (quartzjer) [reply draft-jmiller-json-proof-algorithms.md:159]: Definitely, excellent improvement plan.

File: draft-jmiller-json-web-proof.md [hunk 5]
```diff
@@ -321,30 +321,28 @@ The third payload is the string `"jaydoe@example.org"` with the octet sequence o
...
 ```
 
-The Issuer generates the initial JWP Proof value by then signing the above octet string with their stable key using the same process of creating a JWS with the fixed header, assigning the payload, and extracting the resulting signature.
-
-The final proof value from the issuer is the concatenation of the array of ephemeral signatures along with the final stable signature, then base64url-encoded.
+The final proof value from the Signer is the concatenated array of all of the header and payload signatures, then base64url-encoded.
```
[the remark this answers is not in the document]
Comment (quartzjer) [reply draft-jmiller-json-web-proof.md:345]: Yep!

**Pull request merged**